source/shared/cpp/ObjectModel/MarkDownParser.cpp

```diff
@@ -0,0 +1,425 @@
+#include <vector>
+#include <iomanip>
+#include <iostream>
+#include "MarkDownParser.h"
+
+using namespace AdaptiveCards;
+
+// adjust number of emphasis counts after maching is done
+int MarkDownParser::AdjustEmphasisCounts(int leftOver, std::list<Emphasis>::iterator left_itr, 
+        std::list<Emphasis>::iterator right_itr)
+{
+    int delimCnts = 0;
+    if (leftOver >= 0)
+    {
+        delimCnts = left_itr->m_emphCnts - leftOver;
+        left_itr->m_emphCnts = leftOver;
+        right_itr->m_emphCnts = 0;
+    }
+    else
+    {
+        delimCnts = left_itr->m_emphCnts;
+        right_itr->m_emphCnts = leftOver * (-1);
+        left_itr->m_emphCnts = 0;
+    }
+    return delimCnts;
+}
+
+// Following the rules speicified in CommonMark (http://spec.commonmark.org/0.27/)
+// It generally supports more stricker version of the rules
+std::string MarkDownParser::TransformToHtml()
+{
+    GenSymbolTable ();
+    if (!m_leftLookUpTable.size () || !m_rightLookUpTable.size ())
+    {
+        return "<p>" + m_text + "</p>";
+    }
+
+    std::vector<std::list<Emphasis>::iterator> stk;
```

Comment:
>stk [](start = 47, length = 3)

Give this a more descriptive name please.

Reply:
This comment was marked as resolved but there were no changes.

---
In reply to: [154426600](https://github.com/Microsoft/AdaptiveCards/pull/1003#discussion_r154426600) [](ancestors = 154426600)

Reply:
it was changed at iteration 11.

---
In reply to: [154735329](https://github.com/Microsoft/AdaptiveCards/pull/1003#discussion_r154735329) [](ancestors = 154735329,154426600)

Reply:
Sorry, missed it.

---
In reply to: [154736117](https://github.com/Microsoft/AdaptiveCards/pull/1003#discussion_r154736117) [](ancestors = 154736117,154735329,154426600)